src/Avalonia.Diagnostics/Diagnostics/Controls/BrushEditor.axaml.cs

```diff
@@ -0,0 +1,131 @@
+            clearHandler = (s, e) => Brush = default;
+        }
+
+        protected override Type StyleKeyOverride => typeof(BrushEditor);
+
+        protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
+        {
+            base.OnApplyTemplate(e);
+            if (_cleraButton is not null)
+            {
+                _cleraButton.Click -= clearHandler;
+            }
+            _cleraButton = e.NameScope.Find<Button>("PART_ClearButton");
+            if (_cleraButton is Button button)
+            {
+                button.Click += clearHandler;
+            }
+        }
+
+        /// <summary>
+        ///     Defines the <see cref="Brush" /> property.
+        /// </summary>
+        public static readonly DirectProperty<BrushEditor, IBrush?> BrushProperty =
+                AvaloniaProperty.RegisterDirect<BrushEditor, IBrush?>(
+                    nameof(Brush), o => o.Brush, (o, v) => o.Brush = v);
+
+        private IBrush? _brush;
+
+        public IBrush? Brush
+        {
+            get => _brush;
+            set => SetAndRaise(BrushProperty, ref _brush, value);
+        }
+
+        protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs change)
+        {
+            base.OnPropertyChanged(change);
+
+            if (change.Property == BrushProperty)
+            {
+                if (Brush is ISolidColorBrush scb)
+                {
+                    _colorView.Color = scb.Color;
+                }
+                ToolTip.SetTip(this, Brush?.GetType().Name ?? "(null)");
+                InvalidateVisual();
+            }
+        }
+
+        protected override void OnPointerPressed(PointerPressedEventArgs e)
+        {
+            base.OnPointerPressed(e);
+
+            FlyoutBase.ShowAttachedFlyout(this);
+        }
+
+        public override void Render(DrawingContext context)
+        {
+            base.Render(context);
+
+            var brush = Brush ?? Brushes.Black;
+
+            context.FillRectangle(brush, Bounds);
+
+            var text = (Brush as ISolidColorBrush)?.Color.ToString() ?? "(null)";
+
+            var ft = new FormattedText(text,
+                CultureInfo.CurrentCulture,
+                FlowDirection.LeftToRight,
+                Typeface.Default,
+                10,
+                GetTextBrush(brush));
+
+            context.DrawText(ft,
+                new Point(Bounds.Width / 2 - ft.Width / 2, Bounds.Height / 2 - ft.Height / 2));
+        }
+
+        /// <summary>
+        /// <see cref="https://stackoverflow.com/questions/6763032/how-to-pick-a-background-color-depending-on-font-color-to-have-proper-contrast">How to pick a background color depending on font color to have proper contrast</see>/>
+        /// </summary>
+        /// <param name="brush"></param>
+        /// <returns></returns>
+        private static IBrush GetTextBrush(IBrush brush)
```

Comment:
This should probably use the same algorithm we have for relative luminance. It's what is used in WinUI itself was copied over and used in the ColorPicker itself.

https://github.com/AvaloniaUI/Avalonia/blob/db67ad1f8776b5d1b6d13b139c6e3f464d70e54f/src/Avalonia.Controls.ColorPicker/Helpers/ColorHelper.cs#L26

Used in a very similar way to what you are already doing here.

https://github.com/AvaloniaUI/Avalonia/blob/db67ad1f8776b5d1b6d13b139c6e3f464d70e54f/src/Avalonia.Controls.ColorPicker/Converters/ContrastBrushConverter.cs#L64